gitpod-io/gitpod
# Fix running integration tests as part of build

## Description


This PR makes it possible to use `with-integration-tests` to run integration tests as part of the build job. This functionality has existed for a long time, but it was never updated to work with Harvester-based preview environments.

The old approach was to start a separate Werft job which used the `eu.gcr.io/gitpod-core-dev/build/integration-tests:{{ .Annotations.version }}` image in Werft. This meant it would run the compiled integration tests. However, this image doesn't have any of the utility scripts available we need to configure access to the the preview environment.

So I instead decided to delete the `.werft/run-integration-tests.yaml` job and instead introduce a small helper script - 
 `test/run.sh` - which uses `go` to run the tests. This is similar to the approach adopted by Workspace ([.werft/workspace-run-integration-tests.sh](https://github.com/gitpod-io/gitpod/blob/main/.werft/workspace-run-integration-tests.sh)) and IDE ([.werft/ide-integration-tests-startup.yaml](https://github.com/gitpod-io/gitpod/blob/main/.werft/ide-integration-tests-startup.yaml)). 

My intention is to use `test/run.sh` in both the Workspace and IDE integration test jobs too, so that we have a single way to run integration tests which is used in both Werft and Gitpod workspaces - but I'll look into this in a follow up PR.

---

This PR also adds a new checkbox for `with-integration-tests` to the PR template so spread awareness of the flag.

## Related Issue(s)


Part of https://github.com/gitpod-io/gitpod/issues/12350

## How to test


I ran the script manually from my Gitpod workspace

```sh
./test/run.sh
./test/run.sh ide
./test/run.sh webapp
./test/run.sh workspace
```

I ran it through Werft ([see job](https://werft.gitpod-dev.com/job/gitpod-build-mads-fix-with-integration-tests.46))

```sh
werft job run github -a with-preview="true" -a with-integration-tests="ide"
```

## Release Notes

```release-note
NONE
```

## Documentation

N/A
## Werft options:

   Valid updates are `all`, `workspace`, `webapp`, `ide`
